ui/desktop/electron-app/src/services/client-daemon-manager.js

```diff
@@ -56,12 +55,18 @@ class ClientDaemonManager {
    * Stops the daemon.
    */
   stop() {
-    if (this.#isClientDaemonAlreadyRunning) {
-      return;
+    // We started up the daemon so we should stop it
+    if (!this.#isClientDaemonAlreadyRunning) {
+      const stopDaemonCommand = ['daemon', 'stop'];
+      spawnSync(stopDaemonCommand);
     }
 
-    const stopDaemonCommand = ['daemon', 'stop'];
-    spawnSync(stopDaemonCommand);
+    // Kill the process if it's still running
+    if (this.#clientDaemonProcess && !this.#clientDaemonProcess.killed) {
+      isWindows()
+        ? treeKill(this.#clientDaemonProcess.pid)
+        : this.#clientDaemonProcess.kill();
+    }
```

Comment:
I realized we likely weren't properly killing the process when starting up the client daemon 😓